### ISSUE TITLE
Missing 'Authorization' or 'Cookie' header in JWT authentication mode with Cookie Enabled

### ISSUE THREAD

**Issue by Joan-George** [hasura/graphql-engine#9820]
### Version Information

Server Version:
CLI Version (for CLI related issue): hasura/graphql-engine:v2.29.1

### Environment
Hasura Docker

### What is the current behaviour?

When I enable the Hasura JWT Authentication with header type cookie It's working fine when hit through my application endpoint. But, when I try to add the Cookie in the Hasura console it's not considering the Cookie which I added in the Hasura console header. and throws the error `Missing 'Authorization' or 'Cookie' header in JWT authentication mode`.

### What is the expected behaviour?
Cookies also should be included when making a request from the Hasura console when I add it to the header.

### How to reproduce the issue?

1. In the Hasura docker-compose enable JWT mode with following environment variable.
    `HASURA_GRAPHQL_JWT_SECRET: '{"type":"RS256","jwk_url":"https://cognito-idp.${REGION}.amazonaws.com/${COGNITO_ID}/.well-known/jwks.json","claims_format":"stringified_json","header":{"type":"Cookie","name":"token"}}'`
2. 
3.
4.

### Screenshots or Screencast
![Hasura-console](https://github.com/hasura/graphql-engine/assets/52650773/a916d8e7-f59f-4487-8d9a-120db41ad697)

![no-cookie-request-header](https://github.com/hasura/graphql-engine/assets/52650773/e3d34628-029c-43a4-b59e-4e0a3822dab6)

### Any possible solutions/workarounds you're aware of?

As of now, I don't have any workaround for this.

### Keywords

Hasura Header, Cookie, JWT token

**Comment by manasag** [comment]
Maybe related issue #9818

**Comment by Joan-George** [comment]
Issue #9818 he is trying to set up the subscription with Apollo. But, In my case, I'm adding the header in the Hasura console and I could not see the cookies in the headers of the request.

**Comment by quytran0311** [comment]
I am also facing the same problem, did you fix it? @Joan-George

**Comment by Joan-George** [comment]
No, I have not fixed it yet, But it's taking the cookie from the domain which persists in the same browser. 

In that way, It's working fine. 

In case you are enabling cookie-based authentication you should have an application that sets your token in the cookie when you log in to your application and you have to use the Hasura console in the same browser.

I'm not sure if the working mechanism I mentioned here is the correct way or not. But, It solves the problem.
